src/vcpkg/install.cpp

```diff
@@ -662,16 +704,11 @@ namespace vcpkg::Install
                     const auto find_package_name = Path(path.parent_path()).filename().to_string();
                     if (!ec)
                     {
-                        std::sregex_iterator next(contents.begin(), contents.end(), cmake_library_regex);
-                        std::sregex_iterator last;
-
-                        while (next != last)
+                        auto targets = get_cmake_add_library_names(contents);
+                        if (!targets.empty())
                         {
-                            auto match = *next;
-                            auto& targets = library_targets[find_package_name];
-                            if (std::find(targets.cbegin(), targets.cend(), match[1]) == targets.cend())
-                                targets.push_back(match[1]);
-                            ++next;
+                            auto& all_targets = library_targets[find_package_name];
+                            all_targets.insert(all_targets.end(), targets.begin(), targets.end());
```

Comment:
```suggestion
                            all_targets.insert(all_targets.end(), std::make_move_iterator(targets.begin()), std::make_move_iterator(targets.end()));
```

Needs `#include <iterator>` if not already done in this file.

Reply:
(Also, I hate every map::operator[] I've ever seen and this is no exception but that's not changed here so :P )

Reply:
you will get no disagreement from me